### PR TITLE
Seperate SPD config data from memory config data

### PR DESCRIPTION
**Pull request by raghavag**
This patch will seperate SPD config data from memory configuration data,
as a small change in the SPD config data is resulting in duplication
of the memory config data. After sperating SPD config data, duplication
of whole memory cfg data is not required and would result in saving space.

Signed-off-by: Raghava Gudla <raghava.gudla@intel.com>

### CODE REVIEW

File: Platform/CoffeelakeBoardPkg/Library/Stage1BBoardInitLib/Stage1BBoardInitLib.c [hunk 1]
```diff
@@ -111,10 +114,25 @@ UpdateFspConfig (
       PlatformId == PLATFORM_ID_CFL_S) {
     CopyMem (&Fspmcfg->SpdAddressTable, MemCfgData->SpdAddressTable, sizeof(MemCfgData->SpdAddressTable));
   } else {
-    Fspmcfg->MemorySpdPtr00       = (UINT32) MemCfgData->MemorySpdPtr00;
-    Fspmcfg->MemorySpdPtr01       = (UINT32) MemCfgData->MemorySpdPtr01;
-    Fspmcfg->MemorySpdPtr10       = (UINT32) MemCfgData->MemorySpdPtr10;
-    Fspmcfg->MemorySpdPtr11       = (UINT32) MemCfgData->MemorySpdPtr11;
+    MemSpd0CfgData = (MEM_SPD0_CFG_DATA *)FindConfigDataByTag (CDATA_MEM_SPD0_TAG);
```
Comment (mauricema) [inline Platform/CoffeelakeBoardPkg/Library/Stage1BBoardInitLib/Stage1BBoardInitLib.c]: consider using a loop to simply the code and strings.

File: Platform/CoffeelakeBoardPkg/CfgData/Template_Spd.dsc [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+##
+  #
+  # ================================= SPD_TMPL =========================================
+  #
+    =======================================================================================
+  # !BSF DEFT:{SPD_TMPL:START}
+    # !BSF NAME:{Spd Data CH$(1) DIMM$(2)} TYPE:{Combo}
+    # !BSF OPTION:{0: NONE, 1: SPD0, 2: SPD1}
+    # !BSF HELP:{Select which SPD data should be used for CH$(1) DIMM$(2).SPD0 and SPD1 are associated with configurable SPD data block from MEM SPD}
+    gCfgData.SpdDataSel$(1)$(2)                     |      * | 0x04 | $(3)
```
Comment (mauricema) [inline Platform/CoffeelakeBoardPkg/CfgData/Template_Spd.dsc]: do you need 4 bytes for  this option ?   1 byte should be enough.

File: Platform/CoffeelakeBoardPkg/CfgData/Template_CfgData.dsc [hunk 3]
```diff
@@ -227,3 +227,14 @@
   # ---------------------------------------------------------------------------------------
   # RP_TMPL
   !include Template_PcieRp.dsc
+
+  # ---------------------------------------------------------------------------------------
+  # SPD_TMPL
+  !include Template_Spd.dsc
+
+  # ---------------------------------------------------------------------------------------
+  # MEM_TMPL
```
Comment (mauricema) [inline Platform/CoffeelakeBoardPkg/CfgData/Template_CfgData.dsc]: MEM_TMPL  is only used once,  why do we need a template here ?

Comment (raghavag) [reply Platform/CoffeelakeBoardPkg/CfgData/Template_CfgData.dsc]: Removed MEM_TMPL

File: Platform/CoffeelakeBoardPkg/CfgData/Template_Spd.dsc (hunk 2 above)
Comment (raghavag) [reply Platform/CoffeelakeBoardPkg/CfgData/Template_Spd.dsc]: Changed to 1 byte

File: Platform/CoffeelakeBoardPkg/Library/Stage1BBoardInitLib/Stage1BBoardInitLib.c (hunk 1 above)
Comment (raghavag) [reply Platform/CoffeelakeBoardPkg/Library/Stage1BBoardInitLib/Stage1BBoardInitLib.c]: Did not use loop but reduced code to couple of lines.

**Pull request merged**